tests/decap/test_decap.py

```diff
@@ -0,0 +1,242 @@
+import json
+import logging
+from datetime import datetime
+
+import pytest
+from jinja2 import Template
+from netaddr import IPNetwork
+from ansible.plugins.filter.core import to_bool
+
+from ptf_runner import ptf_runner
+from common.plugins.fib import generate_routes
+
+logger = logging.getLogger(__name__)
+
+PTFRUNNER_QLEN = 1000
+FIB_INFO_DEST = "/root/fib_info.txt"
+
+
+def get_uplink_ports(topology, topo_type):
+
+    uplink_ports = []
+    if topo_type == "t0":
+        for k, v in topology["VMs"].items():
+            if "T1" in k:
+                uplink_ports.append("[{}]".format(" ".join([str(vlan) for vlan in v["vlans"]])))
```

Comment:
'vlans' are concept in fanout and/or deployment procedures. When we move on to future testbed v2, the vlans on the fanaout switch would get auto assigned. It would be better for this attribute name to be indices instead of vlans. But this is apparently out of the scope of this PR.

Reply:
realized that vlan is what we defined in topology file. I see the origin of this vlans tag.

Reply:
Yes, `vlans` is from the topology file. Indeed it is indices of port.